Software/bsw/static/Infrastructure/platform/src/can_lib.c

```diff
@@ -282,19 +283,19 @@ _CANDataRegRead(uint8 *pui8Data, uint32 *pui32Register, uint32 ui32Size)
         // Read out the data 16 bits at a time since this is how the registers
         // are aligned in memory.
         //
-        ui32Value = HWREG(pui32Register++);
+        ui16Value =(uint16) HWREG(pui32Register++);
 
         //
         // Store the first byte.
         //
-        pui8Data[ui32Idx++] = (uint8)ui32Value;
+       *pui8Data++ = (uint8)ui16Value;
```

Comment:
using this instruction this way (which is pointer arithmetic) affects the instruction at line 298 " pui8Data[ui32Idx++] = (uint8)(ui16Value >> 8); " as pui8Data is updated to be the address of the next element (not the address of first element anymore not the array beginning anymore) 
Then we have to use pui8Data in the two instructions in one either ways : accessing array by index(which is easier and more readable then this way is preferred ) or pointer arithmetic, then it should be " pui8Data[ui32Idx++] = (uint8)ui32Value; " then modify the other instruction to be like " pui8Data[ui32Idx++] = (uint8)(ui32Value >> 8);"

Reply:
" as pui8Data is updated to be the address of the next element"
Yes I want to affect the next element to the first of the array because it's the second byte 
*pui8Data++  is the same as pui8Data[ui32Idx++]
it's not an error but I can change it if you think it's more readable

Reply:
of course not an error, but in the first iteration after we store the first byte the pui8Data is pointing to the second element in the array, in the next iteration before we store the 3rd byte pui8Data is still pointing to the second element in the array, then we overwrite the 2nd byte stored in the second element with the 3rd byte , then bugs happen every iteration and we do not know what the wrong is.

Reply:
I see it's because I used ui32Idx only in the second increment
pui8Data[ui32Idx++] = (uint8_t)(ui32Value >> 8); 
thanks .. I'll fix it